lib/phoenix/socket/transport.ex

```diff
@@ -388,6 +388,33 @@ defmodule Phoenix.Socket.Transport do
     end
   end
 
+  def extract(conn, opts) do
```

Comment:
What do you think about including a short documentation? 

Something like:

```
Extracts metadata from the transport.

## Options

    * `:peer_data` - the request peer data (if one is present)
    * `:x_headers` - custom proprietary headers, denoted by the `X-` prefix

```

Reply:
I would `@doc false` it in this case since it's only here to be shared across the WS and LongPoll transport, not used by others. I also think naming it `extract_metadata` is the way to go since we're down below user land